pkg/controller/servicebindingrequest/secret.go

```diff
@@ -46,16 +49,33 @@ func (s *secret) createOrUpdate(payload map[string][]byte) (*unstructured.Unstru
 	resourceClient := s.buildResourceClient()
 
 	logger.Debug("Attempt to create secret...")
-	_, err = resourceClient.Create(u, metav1.CreateOptions{})
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return nil, err
-	}
-
-	logger.Debug("Secret already exists, updating contents instead...")
-	_, err = resourceClient.Update(u, metav1.UpdateOptions{})
+	existingSecret, err := resourceClient.Get(s.name, metav1.GetOptions{})
```

Comment:
`s.get()` can be used instead.

Reply:
Done https://github.com/redhat-developer/service-binding-operator/pull/523/commits/123bc03e78ad4ffa0645ffeaac3854a620203ccf